roles/common/tasks/main.yml

```diff
@@ -16,11 +14,9 @@
 
 - name: common | install building software and build essentials
   ansible.builtin.apt:
-    name: '{{ common_packages }}'
+    name: "{{ item }}"
     state: present
-    update_cache: true
-  ignore_errors: true
-  changed_when: false
+  with_items: "{{ common_packages }}"
```

Comment:
No issue with this, but just curious.  Why did you change the apt command to install the packages one at a time? Better failure information?

Reply:
When you run a task as a list, you get feedback on each item in the list, like this:
```
TASK [common : common | install building software and build essentials] ********
changed: [instance] => (item=htop)
changed: [instance] => (item=python-apt)
changed: [instance] => (item=acl)
changed: [instance] => (item=build-essential)
changed: [instance] => (item=git)
changed: [instance] => (item=unzip)
changed: [instance] => (item=zip)
ok: [instance] => (item=python3-pip)
ok: [instance] => (item=python-setuptools)
changed: [instance] => (item=libssl-dev)
changed: [instance] => (item=zlib1g-dev)
changed: [instance] => (item=tmux)
changed: [instance] => (item=vim)
changed: [instance] => (item=wget)
changed: [instance] => (item=silversearcher-ag)
changed: [instance] => (item=jq)
```
Which is helpful for diagnosing idempotency issues - you can see which package or module is getting re-installed the second time around. 

I think `with_items` is deprecated in favor of `loop`, but `loop` doesn't give the same granular output. At least, not by default - I will do some research to see if there's a config setting to control this.